sql/wsrep_trans_observer.h

```diff
@@ -438,6 +442,7 @@ static inline int wsrep_after_rollback(THD *thd, bool all) {
 
 static inline int wsrep_before_statement(THD *thd) {
   // WSREP_DEBUG("wsrep_before_statement %u", thd->thread_id());
+  thd->wsrep_stmt_transaction_rolled_back = false;
```

Comment:
before_statement is not called before each statement executes. Trace the flow of before the statement. We expect the flag to get cleared before the next transaction starts.